cve_bin_tool/cli.py

```diff
@@ -219,48 +223,42 @@ def main(argv=None):
         )
 
     # Single-thread mode
-    # if not args.multithread:
-
-    # Close database when done
-    cvedb = CVEDB()
-    cvedb.open()
-    with cvedb:
-        scanner = Scanner(cvedb, args.extract, score=score)
-        scanner.remove_skiplist(skips)
-        LOGGER.info(scanner.print_checkers())
-        scanner.recursive_scan(args.directory)
-
-        LOGGER.info("")
-        LOGGER.info("Overall CVE summary: ")
-        LOGGER.info(
-            f"There are {scanner.files_with_cve} files with known CVEs detected"
-        )
-        if scanner.files_with_cve > 0:
-            affected_string = ", ".join(
-                map(
-                    lambda module_version: "".join(str(module_version)),
-                    scanner.affected(),
-                )
-            )
-            LOGGER.info(f"Known CVEs in {affected_string}:")
-
-            # Creates a Object for OutputEngine
-            output = OutputEngine(
-                products=scanner.all_cves,
-                filename=args.output_file,
-                files_with_cve=scanner.files_with_cve,
-                files_without_cve=scanner.files_without_cve,
-                total_files=scanner.total_scanned_files,
+    scanner = VersionScanner(args.extract, score=score)
+    scanner.remove_skiplist(skips)
+    LOGGER.info(scanner.print_checkers())
+    scanner.recursive_scan(args.directory)
+
+    LOGGER.info("")
+    LOGGER.info("Overall CVE summary: ")
+    LOGGER.info(f"There are {scanner.files_with_cve} files with known CVEs detected")
+    if scanner.files_with_cve > 0:
+        affected_string = ", ".join(
+            map(
+                lambda module_version: "".join(str(module_version)), scanner.affected(),
             )
+        )
+        LOGGER.info(f"Known CVEs in {affected_string}:")
+
+        # Creates a Object for OutputEngine
+        output = OutputEngine(
+            products=scanner.all_cves,
+            filename=args.output_file,
+            files_with_cve=scanner.files_with_cve,
+            files_without_cve=scanner.files_without_cve,
+            total_files=scanner.total_scanned_files,
+        )
 
-            if not args.quiet or args.output_file != sys.stdout:
-                output.output_file(args.format)
+        if not args.quiet:
+            output.output_file(args.format)
 
-        # Use the number of files with known cves as error code
-        # as requested by folk planning to automate use of this script.
-        # If no files found, then the program exits cleanly.
-        return scanner.files_with_cve
+    # Use the number of files with known cves as error code
+    # as requested by folk planning to automate use of this script.
+    # If no files found, then the program exits cleanly.
+    return scanner.files_with_cve
 
 
 if __name__ == "__main__":
-    sys.exit(main())
+    if os.getenv("ACTIONS"):
```

Comment:
if we return no. of CVEs as exit status, action will fail if directory contains CVEs. It wasn't failing until now since dir we were scanning doesn't had any CVE. Though I wonder do we really need to provide no. of cves via exit status?